initproducerid.go

```diff
@@ -0,0 +1,86 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/segmentio/kafka-go/protocol"
+	"github.com/segmentio/kafka-go/protocol/initproducerid"
+)
+
+// InitProducerIDRequest is the request structure for the InitProducerId function
+type InitProducerIDRequest struct {
+	// Address of the kafka broker to send the request to.
+	Addr net.Addr
+
+	// The transactional id key.
+	TransactionalID string
+
+	// Time after which a transaction should time out
+	TransactionTimeoutMs int32
+}
+
+// ProducerSession is
+type ProducerSession struct {
+	IsTransactional bool
```

Comment:
It doesn't appear that this is a part of the kafka API?

Reply:
Yes. Will get rid of it. Doesn't need to be exposed.